addons/purchase/models/purchase_order.py

```diff
@@ -1085,3 +1085,17 @@ def _is_readonly(self):
         """
         self.ensure_one()
         return self.state == 'cancel'
+
+    def action_cancel_wizard(self):
+        for order in self:
+            if order.state in ['draft', 'sent', 'purchase']:
```

Comment:
If we only include orders that are all in the done/to approve/cancel state in our selection and try to cancel, there will be no feedback to the user. Maybe we can drop the for loop and if statement here? As we are already checking the cancel conditions in the wizard itself